var/spack/repos/builtin/packages/ffb/package.py

```diff
@@ -0,0 +1,163 @@
+            m.write('make lib\n')
+        os.chmod(makeall, 0o755)
+
+        makeall = join_path(workdir, 'util',  'makeall')
+        les3d_dir = join_path(workdir, 'util', 'les3d.mpi')
+        les3c_dir = join_path(workdir, 'util', 'les3c.mpi')
+        les3ct_dir = join_path(workdir, 'util', 'les3ct.mpi')
+        les3x_dir = join_path(workdir, 'util', 'les3x.mpi')
+        with open(makeall, 'w') as m:
+            m.write('#!/bin/csh -f\n')
+            m.write('setenv LES3DHOME {0}\n'.format(''.join(workdir)))
+            m.write('cd {0}\n'.format(les3d_dir))
+            m.write('make\n')
+            m.write('cd {0}\n'.format(les3c_dir))
+            m.write('make\n')
+            m.write('cd {0}\n'.format(les3ct_dir))
+            m.write('make\n')
+            m.write('cd {0}\n'.format(les3x_dir))
+            m.write('make\n')
+            for d in [les3c_dir, les3ct_dir, les3d_dir]:
+                editfile = join_path(d, 'FILES')
+                m = FileFilter(editfile)
+                m.filter(r'-lmpi_f77', '')
+        os.chmod(makeall, 0o755)
+
+        editfile = join_path(workdir, 'lib', 'src',
+                             'REVOCAP_Refiner-0.4.3', 'OPTIONS')
+        m = FileFilter(editfile)
+        m.filter(r'ARCH\s*=.*$', 'ARCH=`arch`-linux')
+        m.filter(r'CC\s*=.*$', 'CC={0}'.format(spack_cc))
+        m.filter(r'CFLAGS\s*=.*$', 'CFLAGS={0}'.format(' '.join(cflags)))
+        m.filter(r'CXX\s*=.*$',  'CXX={0}'.format(spack_cxx))
+        m.filter(r'CXXFLAGS\s*=.*$',
+                 'CXXFLAGS={0}'.format(' '.join(cxxflags)))
+        m.filter(r'F90\s*=.*$', 'CC={0}'.format(spack_fc))
+        m.filter(r'FFLAGS\s*=.*$', 'FFLAGS={0}'.format(' '.join(cflags)))
+        m.filter(r'LD\s*=.*$', 'LD={0}'.format(spack_fc))
+        m.filter(r'LDFLAGS\s*=.*$',
+                 'LDFLAGS={0}'.format(' '.join(fflags)))
+        m.filter(r'LDSHARED\s*=.*$',
+                 'LDSHARED={0}'.format(' '.join(ldshared)))
+        m.filter(r'LIBS\s*=.*$', 'LIBS={0}'.format(' '.join(libs)))
+        m.filter(r'LIBPATH\s*=.*$', 'LIBPATH= ')
+
+        editfile = join_path(workdir, 'util', 'xvx2gf', 'Makefile')
+        m = FileFilter(editfile)
+        m.filter(
+            r'#LES3DHOME   =', 'LES3DHOME= {0}\n'.format(workdir))
+        m.filter(r'g\+\+', (spack_cxx))
+
+        editfile = join_path(workdir, 'util', 'les3x.mpi', 'FILES')
+        m = FileFilter(editfile)
+        m.filter(r'LIBS = -lfort -lgf2 -ldd_mpi -lmpi_f77',
+                 'LIBS = -lfort -lgf2  -ldd_mpi')
+
+        makeall = Executable(join_path(workdir, 'make',  'Makeall'))
+        makeall()
+
+        make = join_path(workdir, 'make', 'makefile')
+        m = FileFilter(make)
+        m.filter(r'OPTION.spack', 'OPTION\n')
+        makeall = Executable(join_path(workdir,
+                             'lib',  'src', 'dd_mpi', 'Makeall'))
+        makeall()
+
+        makeall = Executable(join_path(workdir, 'util', 'makeall'))
+        makeall()
```

Comment:
This appears to be the first code that would not be better placed in an `edit` phase. This should be the `build` phase